### PR TITLE
Unify r/fbeta/fbeta_score

### LINKED ISSUE #729: Unify functional metrics names
## 🚀 Feature

let's unify the names of the metrics across the package and use `sklearn` and good [reference](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.metrics) 

### Motivation

let's apply the same assumption for all domains and preferably use fully expanded names (as a new person to the domain may not know the particular abbreviation)

### Pitch

be consistent across domains and close to standard

### Alternatives

each of these can be taken as a separate PR

- [ ] `f1` -> `f1_score`
- [ ] `fbeta` -> `fbeta_score`
- [ ] `hinge` -> `hinge_loss`
- [ ] `pit` -> `permutation_invariant_training`
- [ ] `pit_permutate` -> ???
- [ ] `psnr` -> `peak_signal_noise_ratio`
- [ ] `sdr` -> `signal_distortion_ratio` #711
- [ ] `si_sdr` -> `scale_invariant_signal_distortion_ratio` #711
- [ ] `snr` -> `signal_noise_ratio` #712
- [ ] `si_snr` -> `scale_invariant_signal_noise_ratio` #712
- [ ] `ssim` -> `structual_similarity_index_measure`
- [x] `wer` -> `word_error_rate` #714

### PR DESCRIPTION
**Pull request by cuent**
## What does this PR do?

Unify r/fbeta/fbeta_score - part of #729 

## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: torchmetrics/functional/classification/f_beta.py [hunk 1]
```diff
@@ -244,6 +244,30 @@ def fbeta(
...
+    threshold: float = 0.5,
+    top_k: Optional[int] = None,
+    multiclass: Optional[bool] = None,
+) -> Tensor:
+    r"""
+    Computes f_beta metric.
+
+    .. deprecated:: v0.7
+        Use :class:`torchmetrics.functional.f1_score`. Will be removed in v0.8.
+    """
```
Comment (Borda) [inline torchmetrics/functional/classification/f_beta.py:264]:
```suggestion
        Example:
            >>> fbeta(torch.tensor([0, 2, 1, 0, 0, 1]), torch.tensor([0, 1, 2, 0, 1, 2]), num_classes=3, beta=0.5)
            tensor(0.3333)
    """
```

File: torchmetrics/classification/f_beta.py [hunk 2]
```diff
@@ -175,7 +175,47 @@ def compute(self) -> Tensor:
...
 
 
-class F1Score(FBeta):
+class FBeta(FBetaScore):
+    r"""
+    Computes `F-score`_, specifically:
+
+    .. deprecated:: v0.7
+        Use :class:`torchmetrics.FBetaScore`. Will be removed in v0.8.
+    """
```
Comment (Borda) [inline torchmetrics/classification/f_beta.py:184]:
```suggestion
        Example:
            >>> f_beta = FBetaScore(num_classes=3, beta=0.5)
            >>> f_beta(torch.tensor([0, 2, 1, 0, 0, 1]), torch.tensor([0, 1, 2, 0, 1, 2]))
            tensor(0.3333)
    """
```

**Review by Borda — APPROVED** [review]

**Review by ethanwharris — COMMENTED** [review]
Looks good :smiley: Some suggestions

File: torchmetrics/classification/f_beta.py [hunk 3]
```diff
@@ -175,7 +175,51 @@ def compute(self) -> Tensor:
...
+        mdmc_average: Optional[str] = None,
+        ignore_index: Optional[int] = None,
+        top_k: Optional[int] = None,
+        multiclass: Optional[bool] = None,
+        compute_on_step: bool = True,
+        dist_sync_on_step: bool = False,
+        process_group: Optional[Any] = None,
+        dist_sync_fn: Callable = None,
+    ) -> None:
+        warn("`FBeta` was renamed to `FBetaScore` in v0.7 and it will be removed in v0.8", DeprecationWarning)
```
[the remark this answers is not in the document]
Comment (Borda) [reply torchmetrics/classification/f_beta.py:205]: ok, that I will do just before release...

**Review by ethanwharris — APPROVED** [review]

File: torchmetrics/classification/f_beta.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Borda) [reply torchmetrics/classification/f_beta.py:206]: ok, will do it just before release when we have all ongoing deprecation in place, just for record #744

**Pull request merged**